test/parser.js

```diff
@@ -0,0 +1,77 @@
+import { expect } from 'chai';
+import fury from 'fury';
+import Parser from '../src/parser';
+
+describe('Parser', () => {
+  let parser;
+
+  before(() => {
+    parser = new Parser({ minim: fury.minim });
+    parser.swagger = {
+      consumes: [],
+      produces: [],
+    };
+  });
+
+  context('content types', () => {
+    it('gathers null response type when no examples or produces', () => {
+      const methodValue = {};
+      const examples = {};
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal([null]);
+    });
+
+    it('gathers all example response content types', () => {
+      const methodValue = {};
+      const examples = {
+        'application/json': '',
+        'application/hal+json': '',
+        'application/xml': '',
+      };
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal([
+        'application/json',
+        'application/hal+json',
+        'application/xml',
+      ]);
+    });
+
+    it('gathers first JSON produces without examples', () => {
+      const methodValue = {
+        produces: [
+          'application/json',
+          'application/hal+json',
+        ],
+      };
+      const examples = {};
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal(['application/json']);
+    });
+
+    it('gathers only example response content types with produces', () => {
```

Comment:
This test sounds like if the inputs were

```js
const methodValue = {
  produces: [
    'application/json',
    'application/problem+json',
  ],
};
const examples = {
  'application/hal+json': '',
  'application/vnd.error+json': '',
};
```

then following would be expected:

```
['application/vnd.error+json']
```

I'm inferring that from the word "only". If that's correct, I'd add one more content type to the examples for clarity.

Reply:
Also I'm not sure if I'd like such behavior as it seems a bit counter-intuitive to me. Generally it seems the `examples` have higher priority as a source of truth about content types and this would mean they're sort of influenced by `produces`, not overriding it.

Reply:
I intended the test title (`gathers only example response content types with produces`) as the following:

Gather only example response content types (with produces), i.e, there is some produces too. I can see how this can be unclear, but not sure how to better explain it in such a short amount of test. Do you have some suggestion?

Reply:
Wait, my first reading was wrong. My head mixed `application/problem+json` and `application/vnd.error+json`, thinking it's the same thing. Please let me re-review this part in the context of such a surprising finding 😄

Reply:
@kylef I think when I corrected the neuron connections in my brain, the description feels much more fine then previously. I think it's good to go.